scheduler/services/resource/night_resource_configuration.py

```diff
@@ -2,14 +2,30 @@
 # For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
 from dataclasses import dataclass
-from datetime import date
-from typing import FrozenSet
+from datetime import date, datetime, timedelta
+from typing import FrozenSet, Optional
 
 from .filters import AbstractFilter
 
 from lucupy.minimodel import Resource, Site
 
 
+@dataclass(frozen=True)
+class Fault:
+    id: str
+    start: datetime
+    end: datetime
+    time_loss = timedelta
+    comment: str
+
+@dataclass(frozen=True)
+class EngTask:
+    start: Optional[datetime]
+    end: Optional[datetime]
+    time_loss: Optional[timedelta]
+    comment: str
```

Comment:
Since these have so much in common(ish), I was just wondering it there's any way to make them hierarchical with a common supertype? The `Optional`s though might not less us do it.

Reply:
I thought the same but in reality are not the same. The Optional type I think can be modified now that I'm getting the twilights at the moment of parsing.